python/version_details/version_details.py

```diff
@@ -484,11 +485,25 @@ def download_note_attachments(self, note_id):
 
         :param int note_id: The Note entity id.
         """
-        attachments = self.get_note_attachments(note_id)
-
-        for attachment in attachments:
-            uid = self._data_retriever.request_attachment(attachment)
-            self._attachment_uids[uid] = note_id
+        # We're going to query the list of attachments live, because don't
+        # know if the cached data for the activity stream is up to date. The
+        # reason that might be the case is that a new attachment doesn't
+        # trigger a new activity event, so the cache doesn't know it's out
+        # of date in that regard. It would be great to find a better solution
+        # than not trusting the cache.
+        self._attachment_query_uids[self._data_retriever.execute_find(
+            "Attachment",
+            [[
+                "attachment_links",
+                "in",
+                {"type":"Note", "id":note_id}
+            ]],
+            fields=[
+                "this_file",
+                "image",
```

Comment:
Unless those expire nearly instantly it should never be a problem. Since we're querying the list of attachment entities live each time the download is requested we should be getting valid S3 urls.